mmdet/models/layers/transformer/detr_transformer.py

```diff
@@ -263,16 +252,10 @@ def __init__(self,
                      act_cfg=dict(type='ReLU', inplace=True),
                  ),
                  norm_cfg: OptConfigType = dict(type='LN'),
-                 init_cfg: OptConfigType = None,
-                 batch_first: bool = False) -> None:
+                 init_cfg: OptConfigType = None) -> None:
```

Comment:
```
if "batch_first" not in  self.self_attn_cfg:
    self.self_attn_cfg["batch_first"] = True
else:
   assert self.self_attn_cfg["batch_first"] == True, "First dimension of all DETRs in mmdet is `batch`, please set `batch_first`  flag"
```

Reply:
please add this assert for all configs for pytorch's attention

Reply:
> 

Should we set `self.cross_attn_cfg["batch_first"] = True`? After all, `MultiheadAttention` and `MultiScaleDeformableAttention` both need the `batch_first` flag.

Reply:
> s

ok

Reply:
> please add this assert for all configs for pytorch's attention

Maybe not only pytorch's attention? MultiScaleDeformableAttention needs this assert too.